calico/felix/ipsets.py

```diff
@@ -149,9 +145,30 @@ def on_tags_update(self, profile_id, tags):
             None if deleted.
         """
         _log.info("Tags for profile %s updated", profile_id)
-        old_tags = self.tags_by_prof_id.get(profile_id, [])
-        new_tags = tags or []
-        self._process_tag_updates(profile_id, set(old_tags), set(new_tags))
+
+        # General approach is to default to the empty list if the new/old
```

Comment:
I think the top level comment on this routine should say what the routine does and why - i.e. that it is called when the tags in a profile have changed, and hence the content of the ipset must change.

That sounds obvious, but it took me a moment to figure out and an extra one line comment in each method of this type would be helpful here.